apps/editing-toolkit/editing-toolkit-plugin/wpcom-universal-themes/index.php

```diff
@@ -0,0 +1,292 @@
+ * always returning false when `gutenberg_is_fse_theme` is false.
+ *
+ * @return boolean Core FSE is active.
+ */
+function is_core_fse_active() {
+	// If `gutenberg_is_fse_theme` is false, this is a non-starter.
+	if ( ! is_fse_theme() ) {
+		return false;
+	}
+
+	// Now we just check for our own option.
+	return (bool) get_option( OPTION_NAME );
+}
+
+/**
+ * Proxy for `gutenberg_is_fse_theme` with `function_exists` guarding.
+ *
+ * @uses gutenberg_is_fse_theme
+ *
+ * @return boolean
+ */
+function is_fse_theme() {
+	return function_exists( 'gutenberg_is_fse_theme' ) && gutenberg_is_fse_theme();
+}
+
+/**
+ * Activates Core FSE by setting our option.
+ * Note that even setting this option to true will make no difference on a classic theme.
+ *
+ * @return void
+ */
+function activate_core_fse() {
+	update_option( OPTION_NAME, true );
+}
+
+/**
+ * Deactivates Core FSE by removing our option.
+ *
+ * @return void
+ */
+function deactivate_core_fse() {
+	delete_option( OPTION_NAME );
+}
+
+/**
+ * Hook and unhook Gutenberg's FSE things.
+ *
+ * @return void
+ */
+function load_core_fse() {
+	// perfect parity would put the admin notices back but we don't want that.
+	add_action( 'admin_menu', 'gutenberg_site_editor_menu', 9 );
+	add_action( 'admin_menu', 'gutenberg_remove_legacy_pages' );
+	add_action( 'admin_bar_menu', 'gutenberg_adminbar_items', 50 );
+	add_filter( 'menu_order', 'gutenberg_menu_order' );
+	remove_action( 'init', __NAMESPACE__ . '\hide_template_cpts', 11 );
+	remove_action( 'restapi_theme_init', __NAMESPACE__ . '\hide_template_cpts', 11 );
+}
+
+/**
+ * Loads our menus
+ *
+ * @return void
+ */
+function load_helpers() {
+	// we don't need to show anything to non-FSE-capable themes.
+	if ( ! is_fse_theme() ) {
+		return;
+	}
+	if ( apply_filters( 'a8c_hide_core_fse_activation', false ) ) {
+		return;
+	}
+	add_action( 'admin_notices', __NAMESPACE__ . '\theme_nag' );
+	add_action( 'admin_menu', __NAMESPACE__ . '\add_submenu' );
+	add_action( 'admin_init', __NAMESPACE__ . '\init_settings' );
+}
+
+/**
+ * Unloads our menus
+ *
+ * @return void
+ */
+function unload_helpers() {
+	remove_action( 'admin_notices', __NAMESPACE__ . '\theme_nag' );
+	remove_action( 'admin_menu', __NAMESPACE__ . '\add_submenu' );
+	remove_action( 'admin_init', __NAMESPACE__ . '\init_settings' );
+}
+
+/**
+ * Adds our submenu
+ *
+ * @return void
+ */
+function add_submenu() {
+	add_theme_page(
+		__( 'Site Editor (beta)', 'full-site-editing' ),
+		sprintf(
+		/* translators: %s: "beta" label. */
+			__( 'Site Editor %s', 'full-site-editing' ),
+			'<span class="awaiting-mod">' . esc_html__( 'beta', 'full-site-editing' ) . '</span>'
+		),
+		'edit_theme_options',
+		'site-editor-toggle',
+		__NAMESPACE__ . '\menu_page'
+	);
+}
+
+/**
+ * Prints an admin notice on the themes screen when an FSE theme is active and
+ * this plugin's toggle is inactive. Links to our submenu for activation.
+ *
+ * @return void
+ */
+function theme_nag() {
```

Comment:
Where am I supposed to see this notice? It's not popping up on the themes:

![image](https://user-images.githubusercontent.com/2256104/126649237-4a213880-e390-4117-ac72-fb56fc256289.png)

Reply:
Ah! It only show up at `wp-admin/themes.php`, in wp-admin. Once we're ready to allow all users to opt in, we can also put the notice in Calypso.

Reply:
Although, we might handle the notice differently here, perhaps prompting someone instead while they're activating an FSE theme.